railties/lib/rails/application/configuration.rb

```diff
@@ -51,6 +51,14 @@ def initialize(*)
         @x                             = Custom.new
       end
 
+      def static_cache_control=(value)
+        ActiveSupport::Deprecation.warn("static_cache_control is deprecated and will be removed in Rails 5.1. " \
+                                        "Please use `config.public_file_server.headers['Cache-Control'] = #{value} instead.")
+
+        @static_cache_control = value
+        public_file_server.headers['Cache-Control'.freeze] = value
```

Comment:
This is a bit smelly, but somewhere we have to this to avoid duplicate warnings from `ActionDispatch::Static`.